### PR TITLE
fix issue # 643

### LINKED ISSUE #643: Historic Route/Schedules: Download Transitfeed GTFS
Write a script that, given a date and agency, downloads the appropriate GTFS from TransitFeeds and uploads it to S3. It doesn't download anything from TransitFeeds if the GTFS is already in S3. This script would be deployed to run before save_routes does.

Is an extension of #642 but could be done concurrently

### PR DESCRIPTION
**Pull request by architec**
Fixes #643

### CODE REVIEW

**Comment by arctdav** [comment]
@EddyIonescu

**Review by EddyIonescu — COMMENTED** [review]
Thanks for making the PR and sorry for the delay in reviewing it; I'll be sure to be more responsive over the next few days (also since it's a long-weekend) for reviewing your changes. 

So there are two issues:
1. While it downloads the feed, it doesn't find the feed, which was part of the ticket. It's fine if you want to leave it as a TODO for another PR. For example, in backend/agencies you'd add a "transitfeed_url" to the yaml file of each agency, then you'd make an API call to that URL as to get the feed versions.
2. The goal isn't to always get the latest version but to get the latest version as of a given date. For example, if the date given is 2020-04-01 and there are GTFS feeds from 2020-05-01, 2020-03-01, and 2020-01-01, we'd want the one from 2020-03-01. Same as above, it's fine if you want to leave it as a TODO for another PR.
From the issue description, the start function should take in two arguments: an agency ID (name of the agency that corresponds to the yaml files in backend/agencies, like ttc or muni).

File: backend/download_transitfeed_gtfs.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    return False
+
+def download_upload_S3(transitfeed_key, feedID, S3bucket_name):
+    """
+    download GTFS file fomr transitfeed, then upload downloaded file to
+    S3 bucket, and delete the downloaded file locally
+    :param transitfeed_key: transitfeeds API key
+    :param feedID: feed ID from transitfeed
+    :param S3bucket_name: S3 bucket name
+    :return: True if file was uploaded, else False
```
[the remark this answers is not in the document]
Comment (arctdav) [reply backend/download_transitfeed_gtfs.py:85]: @EddyIonescu could you set this as another PR?

**Comment by arctdav** [comment]
@EddyIonescu, could you make point #1 and #2 in your comment as another PR? I will get back to it when I have time if no one else is interested.

File: backend/download_transitfeed_gtfs.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+
+
+def upload_file(file_name, bucket, object_name=None):
+    """Upload a file to an S3 bucket, then delete this file 
+
+    :param file_name: File to upload
+    :param bucket: Bucket to upload to
+    :param object_name: S3 object name. If not specified then file_name is used
+    :return: True if file was uploaded, else False
```
[the remark this answers is not in the document]
Comment (EddyIonescu) [reply backend/download_transitfeed_gtfs.py]: Clarifying this @arctdav, the try/except wouldn't be needed either (see original comment). Also don't worry about resolving conversations as GitHub automatically buries them when you update your PR; the issue is that resolving conversations makes it hard to see past comments.

File: backend/download_transitfeed_gtfs.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    s3_client = boto3.client('s3')
+    try:
+        response = s3_client.upload_file(file_name, bucket, object_name)
+    except ClientError as e:
+        logging.error(e)
+        return False
+    
+    #* delete this file after upload
+    p = pathlib.Path('./{}'.format(file_name))
+    p.unlink()
```
[the remark this answers is not in the document]
Comment (EddyIonescu) [reply backend/download_transitfeed_gtfs.py:56]: Why was this resolved?

**Comment by EddyIonescu** [comment]
@arctdav the issue with not doing #1 and #2 is that then there's nothing to really use the script for. If another contributor picks up the ticket up they can start off from your branch.

**Review by hathix — CHANGES_REQUESTED** [review]
Please make changes based on Eddy's feedback.

**Pull request closed without merging**